src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs

```diff
@@ -287,8 +287,20 @@ public static int Main()
     }
 }
 ";
-
-            ParseAndValidate(test, TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6));
+            ParseAndValidate(test, TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6),
+                // (7,9): error CS8058: Feature 'tuples' is experimental and unsupported; use '/features:tuples' to enable.
+                //         (a, b) =>
+                Diagnostic(ErrorCode.ERR_FeatureIsExperimental, "(a, b)").WithArguments("tuples", "tuples").WithLocation(7, 9),
+                // (7,16): error CS1001: Identifier expected
+                //         (a, b) =>
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "=>").WithLocation(7, 16),
+                // (7,16): error CS1003: Syntax error, ',' expected
+                //         (a, b) =>
+                Diagnostic(ErrorCode.ERR_SyntaxError, "=>").WithArguments(",", "=>").WithLocation(7, 16),
+                // (7,18): error CS1002: ; expected
+                //         (a, b) =>
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(7, 18)
+                );
```

Comment:
This is evidence that we need to really bang on the parser in situations that involve mixing lambdas, casts, tuple types, tuple expressions, etc.

---
In reply to: [62090097](https://github.com/dotnet/roslyn/pull/11041#discussion_r62090097) [](ancestors = 62090097)